bluesky/protocols.py

```diff
@@ -1,15 +1,25 @@
-try:
-    from typing_extensions import Literal, Protocol, TypedDict, runtime_checkable
-except ImportError:
-    from typing import Literal, Protocol, TypedDict, runtime_checkable
```

Comment:
How did this even work?!

Reply:
https://github.com/bluesky/bluesky/blob/1b0e3bfc623405ef7ab422bc0a9e3036d57a5702/requirements.txt#L10

because of conditional dependencies... guaranteed to have typing_extensions for 3.8, but all available in 3.8+, perfectly fine to use the `typing_extensions` if exists, though

Reply:
I must have typoed something when I was testing it earlier as I thought `from typing import TypedDict` failed 🐑 .

That leads to a different question as to why we are pulling `TypedDict` from `typeing_extensions` rather from `typing`.

Reply:
The one from `typing` does seem to fail. I did a bit of investigation but can't figure out why. @coretl any ideas?

Reply:
Cos of https://github.com/bluesky/bluesky/issues/1572